### PR TITLE
Add CBMC proof for HTTPClient_InitializeRequestHeaders

### PR DESCRIPTION
**Pull request by yourslab**
CBMC proof returns no errors or warnings. However, one thing to note is that the SAT problem had 5755032 variables, 25643058 clauses, taking me hours just to run the proof. I am not sure why this is. There might be a better way to write the proof so as to reduce the runtime.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: libraries/standard/http/cbmc/include/http_cbmc_state.h [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+HTTPRequestHeaders_t * allocateHttpRequestHeaders( HTTPRequestHeaders_t * pRequestHeaders );
+
+/**
+ * @brief Validates if a request headers object is feasible.
+ *
+ * @param[in] pRequestHeaders Request headers to validate.
+ *
+ * @return 1 if request headers is feasible; 0 otherwise.
+ */
+int isValidHttpRequestHeaders( const HTTPRequestHeaders_t * pRequestHeaders );
```
Comment (markrtuttle) [inline libraries/standard/http/cbmc/include/http_cbmc_state.h:52]: int -> bool for predicates.  I may have missed this in your other pull request.

File: libraries/standard/http/cbmc/include/http_cbmc_state.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+HTTPRequestInfo_t * allocateHttpRequestInfo();
+
+/**
+ * @brief Validates if a request info object is feasible.
+ *
+ * @brief param[in] pRequestInfo Request info to validate.
+ *
+ * @return 1 if request headers is feasible; 0 otherwise.
+ */
+int isValidHttpRequestInfo( const HTTPRequestInfo_t * pRequestInfo );
```
Comment (markrtuttle) [inline libraries/standard/http/cbmc/include/http_cbmc_state.h:66]: int -> bool

File: libraries/standard/http/cbmc/proofs/HTTPClient_AddHeader/Makefile [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+DEFINES +=
+INCLUDES +=
+
+REMOVE_FUNCTION_BODY +=
+UNWINDSET += strncmp.0:5
+
+PROOF_SOURCES += $(PROOFDIR)/$(HARNESS_FILE).c 
+PROOF_SOURCES += $(PROOFDIR)/../../sources/http_cbmc_state.c
+
```
Comment (markrtuttle) [inline libraries/standard/http/cbmc/proofs/HTTPClient_AddHeader/Makefile:12]: Might want to delete blank link to be consistent with your other Makefile.

File: libraries/standard/http/cbmc/include/http_cbmc_state.h (hunk 1 above)
Comment (sarenameas) [reply libraries/standard/http/cbmc/include/http_cbmc_state.h:52]: Yea let's go ahead and use bool just for the proofs only.

File: libraries/standard/http/cbmc/proofs/HTTPClient_AddHeader/Makefile (hunk 3 above)
Comment (yourslab) [reply libraries/standard/http/cbmc/proofs/HTTPClient_AddHeader/Makefile:12]: Wow how'd you catch that haha

**Pull request merged**